shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +96,419 @@ public static void addMetaTag(
+        List<Coding> codings = serviceRequest.getCode().getCoding();
+        if (codings.isEmpty()) {
+            return null;
+        }
+        return codings.get(0);
+    }
+
+    public static void setORC4Coding(ServiceRequest serviceRequest, Coding coding) {
+        serviceRequest.getCode().setCoding(List.of(coding));
+    }
+
+    // ORC-4.1 - Entity Identifier
+    public static String getORC4_1Value(ServiceRequest serviceRequest) {
+        Coding coding = getORC4Coding(serviceRequest);
+        if (coding == null) {
+            return null;
+        }
+        return coding.getCode();
+    }
+
+    public static void setORC4_1Value(ServiceRequest serviceRequest, String value) {
+        Coding coding = getORC4Coding(serviceRequest);
+        if (coding == null) {
+            coding = new Coding();
+            setORC4Coding(serviceRequest, coding);
+        }
+        coding.setCode(value);
+    }
+
+    // ORC-4.2 - Namespace ID
+    public static String getORC4_2Value(ServiceRequest serviceRequest) {
+        Coding coding = getORC4Coding(serviceRequest);
+        if (coding == null) {
+            return null;
+        }
+        return coding.getDisplay();
+    }
+
+    public static void setORC4_2Value(ServiceRequest serviceRequest, String value) {
+        Coding coding = getORC4Coding(serviceRequest);
+        if (coding == null) {
+            coding = new Coding();
+            setORC4Coding(serviceRequest, coding);
+        }
+        coding.setDisplay(value);
+    }
+
+    // HD - Hierarchic Designator
+    public static Identifier getHD1Identifier(List<Identifier> identifiers) {
+        return getHl7FieldIdentifier(identifiers, EXTENSION_HD1_DATA_TYPE);
+    }
+
+    // EI - Entity Identifier
+    public static String getEI1Value(Identifier identifier) {
+        return identifier.getValue();
+    }
+
+    public static void setEI1Value(Identifier identifier, String value) {
+        identifier.setValue(value);
+    }
+
+    public static String getEI2Value(Identifier identifier) {
+        return identifier
+                .getExtensionByUrl(EXTENSION_ASSIGNING_AUTHORITY_URL)
+                .getExtensionByUrl(EXTENSION_NAMESPACE_ID_URL)
+                .getValue()
+                .primitiveValue();
+    }
+
+    public static void setEI2Value(Identifier identifier, String value) {
+        identifier
+                .getExtensionByUrl(EXTENSION_ASSIGNING_AUTHORITY_URL)
+                .getExtensionByUrl(EXTENSION_NAMESPACE_ID_URL)
+                .setValue(new StringType(value));
+    }
+
+    protected static Organization createOrganization() {
+        Organization organization = new Organization();
+        String organizationId = UUID.randomUUID().toString();
+        organization.setId(organizationId);
+        return organization;
+    }
+
+    protected static MessageHeader.MessageDestinationComponent createMessageDestinationComponent() {
```

Comment:
This is only called by test code.  Let's remove it.  If you want this as a helper method to make testing easier, add it to (or create new) a helper class that is just inside the `test/groovy/` area.

Reply:
Moved to test package